engine/runner/src/main/scala/org/enso/runner/Main.scala

```diff
@@ -65,6 +65,7 @@ object Main {
   private val IR_CACHES_OPTION               = "ir-caches"
   private val NO_IR_CACHES_OPTION            = "no-ir-caches"
   private val NO_READ_IR_CACHES_OPTION       = "no-read-ir-caches"
+  private val ENABLE_ASSERTIONS_OPTION       = "enable-assertions"
```

Comment:
...should we have `enable-assertions`? Shouldn't we base this value on `-ea` status of the JVM? I believe it would be good enough for our (testing only) purposes. Just:
```java
boolean assertOn = false;
assert assertOn = true;
if (assertOn) {
  // set the `RuntimeOptions.ENABLE_ASSERTIONS`
}
```
Or do we have a really strong reason to expose assertions to Enso CLI users explicitly?

Reply:
I think the issue is, the engine currently violates Truffle assertions - running `test/Tests` with `-ea` fails pretty quickly with:
```
Any.map_nothing:  [2/2, 75ms]
    - should apply the function to a value [62ms]
    - should return `Nothing` unchanged [13ms]
Callables:  [5/5, 100ms]
    - should be able to be applied in a pipeline using |> [44ms]
    - should be able to be applied to an argument using <| [21ms]
    - should be able to be composed backward using << [13ms]
    - should be able to be composed forward using >> [12ms]
    - should define generic inequality on values [8ms]
Any's methods:  [3/3, 670ms]
    - should not be overridable when called statically [220ms]
    - static method calls on Any should have defaulted self argument to Any [131ms]
    - instance methods on different types should not have defaulted self argument [317ms]
Execution finished with an error: java.lang.IllegalArgumentException: Cause exception must extend AbstractTruffleException but was class java.lang.ArrayIndexOutOfBoundsException.
        at <java> org.graalvm.truffle/com.oracle.truffle.api.interop.InteropException.validateCause(InteropException.java:109)
        at <java> org.graalvm.truffle/com.oracle.truffle.api.interop.InteropException.<init>(InteropException.java:56)
        at <java> org.graalvm.truffle/com.oracle.truffle.api.interop.InvalidArrayIndexException.<init>(InvalidArrayIndexException.java:67)
        at <java> org.graalvm.truffle/com.oracle.truffle.api.interop.InvalidArrayIndexException.create(InvalidArrayIndexException.java:126)
        at <java> org.enso.interpreter.runtime.data.vector.ArrayLikeAtNode.arrayAt(ArrayLikeAtNode.java:36)
        at <java> org.enso.interpreter.runtime.data.vector.ArrayLikeAtNodeGen.executeAt(ArrayLikeAtNodeGen.java:71)
        at <java> org.enso.interpreter.node.expression.builtin.immutable.AtVectorNode.execute(AtVectorNode.java:23)
        at <java> org.enso.interpreter.node.expression.builtin.immutable.AtVectorMethodGen.handleExecute(AtVectorMethodGen.java:171)
        at <java> org.enso.interpreter.node.expression.builtin.immutable.AtVectorMethodGen$1Inlineable.call(AtVectorMethodGen.java:91)
        at <java> org.enso.interpreter.node.callable.ExecuteCallNode.callInlineable(ExecuteCallNode.java:63)
        at <java> org.enso.interpreter.node.callable.ExecuteCallNodeGen.executeCall(ExecuteCallNodeGen.java:76)
        at <java> org.enso.interpreter.node.callable.dispatch.CurryNode.doCall(CurryNode.java:155)
        at <java> org.enso.interpreter.node.callable.dispatch.CurryNode.execute(CurryNode.java:108)
        at <java> org.enso.interpreter.node.callable.dispatch.InvokeFunctionNode.invokeCached(InvokeFunctionNode.java:113)
        at <java> org.enso.interpreter.node.callable.dispatch.InvokeFunctionNodeGen.execute(InvokeFunctionNodeGen.java:80)
        at <java> org.enso.interpreter.node.callable.InvokeMethodNode.doFunctionalDispatchCachedSymbol(InvokeMethodNode.java:160)
        at <java> org.enso.interpreter.node.callable.InvokeMethodNodeGen.execute(InvokeMethodNodeGen.java:198)
        at <java> org.enso.interpreter.node.callable.InvokeCallableNode.invokeDynamicSymbol(InvokeCallableNode.java:254)
        at <java> org.enso.interpreter.node.callable.InvokeCallableNodeGen.execute(InvokeCallableNodeGen.java:137)
        at <java> org.enso.interpreter.node.callable.ApplicationNode.executeGeneric(ApplicationNode.java:96)
        at <java> org.enso.interpreter.node.callable.function.BlockNode.executeGeneric(BlockNode.java:54)
        at <java> org.enso.interpreter.node.ClosureRootNode.execute(ClosureRootNode.java:85)
        at <enso> Array.at(Internal)
        at <enso> Array_Spec.test_arrays<arg-0>(src\Data\Array_Spec.enso:24:10-18)
        at <enso> Array_Spec.test_arrays<arg-1>(src\Data\Array_Spec.enso:24:9-58)
        at <enso> Test.run_spec<arg-1>(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Test\0.0.0-dev\src\Test.enso:200:18-25)
        at <enso> Panic.catch(Internal)
        at <enso> Panic.type.recover(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Base\0.0.0-dev\src\Panic.enso:185-190)
        at <enso> Test.run_spec(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Test\0.0.0-dev\src\Test.enso:199-203)
        at <enso> case_branch<arg-3>(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Test\0.0.0-dev\src\Test.enso:77:73-89)
        at <enso> case_branch<arg-1>(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Test\0.0.0-dev\src\Test.enso:77:49-90)
        at <enso> Runtime.no_inline(Internal)
        at <enso> Duration.type.time_execution(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Base\0.0.0-dev\src\Data\Time\Duration.enso:127:18-43)
        at <enso> case_branch(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Test\0.0.0-dev\src\Test.enso:77:24-91)
        at <enso> Test.specify(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Test\0.0.0-dev\src\Test.enso:76-78)
        at <enso> Array_Spec.test_arrays(src\Data\Array_Spec.enso:22-25)
        at <enso> Array_Spec.spec<arg-1>(src\Data\Array_Spec.enso:29:9-35)
        at <enso> case_branch<arg-1>(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Test\0.0.0-dev\src\Test.enso:40:29-37)
        at <enso> case_branch<arg-1>(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Test\0.0.0-dev\src\Test.enso:39-41)
        at <enso> Panic.catch(Internal)
        at <enso> case_branch(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Test\0.0.0-dev\src\Test.enso:38-41)
        at <enso> Test.group<arg-1>(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Test\0.0.0-dev\src\Test.enso:31-46)
        at <enso> Test.group(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Test\0.0.0-dev\src\Test.enso:30-46)
        at <enso> Array_Spec.spec(src\Data\Array_Spec.enso:28-43)
        at <enso> Main.main<arg-1>(src\Main.enso:87:5-19)
        at <enso> Test_Suite.type.run<arg-1>(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Test\0.0.0-dev\src\Test_Suite.enso:69:17-21)
        at <enso> Test_Suite.type.run<arg-1>(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Test\0.0.0-dev\src\Test_Suite.enso:68-70)
        at <enso> Test_Reporter.wrap_junit_testsuites(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Test\0.0.0-dev\src\Test_Reporter.enso:18:14-19)
        at <enso> Test_Suite.type.run(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Test\0.0.0-dev\src\Test_Suite.enso:67-70)
        at <enso> Test_Suite.type.run_main(C:\NBO\enso\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Test\0.0.0-dev\src\Test_Suite.enso:40:13-39)
        at <enso> Main.main(src\Main.enso:85-155)
```

So if the Enso-specific assertions are to be useful for devs, we need to be able to enable them _without_ enabling Java ones. Otherwise, we cannot really do anything.

Ideally, we should indeed enable both. But currently, if I enable both I just cannot do anything useful because the Java asserts will fail too much.

Reply:
Note that I tried to enable Java asserts in these tests in #7882. If the fixes for that are easy, I might merge that PR before this PR. If not, we will need a CMD line argument or an env variable to turn on Enso assertions, while Java assertions will still be disabled.

Reply:
> engine currently violates Truffle assertions

I see. That's an issue. Fixing #7882 would be fine, but we don't have to block this issue by #7882. If #7882 is hard to fix, I'd be fine with `ENSO_ENABLE_ASSERTIONS` - I'd still prefer not to have the CLI option - but it is just my opinion.

Reply:
Removed CMD line option, and added env var setting in b435b75360098870a8973ac600c81472ec623d2e